# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/new-regexp/RegExpAPI.cpp
# Commit: f53110858cb9
# Full Hash: f53110858cb9228ea0c7e056c3f7bc01b5fd76c9
# Author: Iain Ireland <iireland@mozilla.com>
# Date: 2020-05-12 04:04:10
# Regressor Bug: 1634135
# File Overlap Count: 1
# Description:
#   Bug 1634135: Disable named capture parsing until fully supported r=mgaudet
#   
#   The engine supports parsing named captures, but we don't have the code in place to expose the captured groups. Until we do, this code will make sure that we get a syntax error when parsing them.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D73115
# ==============================================================================

diff -r 49b9d6a8a3a5 -r f53110858cb9 js/src/new-regexp/RegExpAPI.cpp
--- a/js/src/new-regexp/RegExpAPI.cpp	Sun May 10 16:00:25 2020 +0000
+++ b/js/src/new-regexp/RegExpAPI.cpp	Sun May 10 16:00:25 2020 +0000
@@ -248,6 +248,14 @@
     ReportSyntaxError(ts, result, chars.begin().get(), chars.length());
     return false;
   }
+  if (!result.capture_name_map.is_null()) {
+    // We can parse named captures, but we don't support them yet.
+    // Until we do, this will report a syntax error:
+    //  "invalid capture group name in regular expression"
+    result.error = RegExpError::kInvalidCaptureGroupName;
+    ReportSyntaxError(ts, result, chars.begin().get(), chars.length());
+    return false;
+  }
   return true;
 }
 
@@ -259,6 +267,14 @@
     ReportSyntaxError(ts, result, pattern);
     return false;
   }
+  if (!result.capture_name_map.is_null()) {
+    // We can parse named captures, but we don't support them yet.
+    // Until we do, this will report a syntax error:
+    //  "invalid capture group name in regular expression"
+    result.error = RegExpError::kInvalidCaptureGroupName;
+    ReportSyntaxError(ts, result, pattern);
+    return false;
+  }
   return true;
 }
 